src/python/pants/pantsd/service/pants_service.py

```diff
@@ -31,12 +31,19 @@ def is_killed(self):
     """
     return self._kill_switch.is_set()
 
-  def setup(self, lock):
+  def setup(self, lifecycle_lock, fork_lock):
     """Called before `run` to allow for service->service or other side-effecting setup.
 
-    :param threading.Lock lock: A global service<->service lock for guarding critical work.
+    :param threading.RLock lifecycle_lock: A lock to guard the service thread lifecycles. This
```

Comment:
Rather than copying the docstring to each of these, just referring to one big docstring somewhere might be good.